src/Components/Components/src/Binding/IFormValueSupplier.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Components.Binding;
+
+/// <summary>
+/// Binds form data valuesto a model.
+/// </summary>
+public interface IFormValueSupplier
```

Comment:
If possible let's keep form stuff in M.A.C.Web. I am aware this differs from query, but querystrings are more universal (e.g., they are relevant for navigation in WebView and even in native Android apps).

Reply:
Ok, I think this can move to Microsoft.AspNet.Core.Components.Web. let me try and do the CascadingValueProvider and this refactor in a separate change, will that work?

Reply:
I'm not totally sure what split you're proposing, but it's kind of tricky to keep track of what's going on if we add a bunch of code to M.A.Components in one PR then have to also remove it in the next one. Hopefully the amount of refactoring in this PR wouldn't be massive. Would it be possible to add the code only in the intended places here?

Reply:
This one is harder to remove, the issue is that it is used by `CascadingModelBinder` that lives in Microsoft.AspNetCore.Components. If we wanted to move it to `Microsoft.AspNetCore.Components.Web` then `Microsoft.AspNetCore.Components.Forms` can't use it.

Would we be happy if we move this to `Microsoft.AspNetCore.Components.Forms` and we give `Microsoft.AspNetCore.Components.Forms` IVT to `Microsoft.AspNetCore.Components`? (Web already has it) or do we also want to keep forms "web agnostic"?

The only other thing that I can think about is having some sort of "interface" for CascadingModelBinder in `Microsoft.AspNetCore.Components`, have some base implementation in `Microsoft.AspNetCore.Components.Forms` and have a concrete implementation in `Microsoft.AspNetCore.Components.Web`

Reply:
The other challenge is that `Route View` wires it up to the page. We could inject a service into `RouteView` to wrap the page being rendered?

Reply:
I've reworked something that might work and I've put it on
https://github.com/dotnet/aspnetcore/tree/javiercn/change-cascading-model-binder

I've filed an issue https://github.com/dotnet/aspnetcore/issues/48454 for us to discuss further, but I am not going to approach it as part of this PR.